azure-spring-boot/src/main/java/com/microsoft/azure/keyvault/spring/KeyVaultEnvironmentPostProcessorHelper.java

```diff
@@ -34,17 +36,24 @@ public KeyVaultEnvironmentPostProcessorHelper(final ConfigurableEnvironment envi
     }
 
     public void addKeyVaultPropertySource() {
-        final String clientId = getProperty(this.environment, Constants.AZURE_CLIENTID);
-        final String clientKey = getProperty(this.environment, Constants.AZURE_CLIENTKEY);
         final String vaultUri = getProperty(this.environment, Constants.AZURE_KEYVAULT_VAULT_URI);
         final Long refreshInterval = Optional.ofNullable(
                 this.environment.getProperty(Constants.AZURE_KEYVAULT_REFRESH_INTERVAL))
                 .map(Long::valueOf).orElse(Constants.DEFAULT_REFRESH_INTERVAL_MS);
         final long timeAcquiringTimeoutInSeconds = this.environment.getProperty(
-                Constants.AZURE_TOKEN_ACQUIRE_TIMEOUT_IN_SECONDS, Long.class, Constants.TOKEN_ACQUIRE_TIMEOUT_SECS);
-
-        final ServiceClientCredentials credentials = new AzureKeyVaultCredential(clientId, clientKey,
+            Constants.AZURE_TOKEN_ACQUIRE_TIMEOUT_IN_SECONDS, Long.class, Constants.TOKEN_ACQUIRE_TIMEOUT_SECS);
+        final ServiceClientCredentials credentials;
+
+        if (this.environment.containsProperty("MSI_ENDPOINT")
+            && this.environment.containsProperty("MSI_SECRET")) {
+            credentials = new AppServiceMSICredentials(AzureEnvironment.AZURE);
```

Comment:
This constructor `AppServiceMSICredentials(AzureEnvironment environment)` is going to load the `MSI_ENDPOINT` and `MSI_SECRET` from `System environment variable`, while the `environment.containsProperty` only promises the properties can be found in the environment, but cannot promise the `System` contains those variables, which may cause potential issues. May need enhancement in another pull request.